src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -133,6 +135,11 @@ OpenTransactionsForAllTasks(List *taskList, int connectionFlags)
 			 * transaction to fail.
 			 */
 			MarkRemoteTransactionCritical(connection);
+
+			if (task->taskType == VACUUM_ANALYZE_TASK)
+			{
+				PQsetNoticeReceiver(connection->pgConn, &VerboseNoticeReceiver, NULL);
```

Comment:
Does this have any side-effects for connections which are cached? (e.g. router query; VACUUM; router query with notice?).